strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
For me leave it as is, I am sure any reader understand it, and I see that in near future we will need to filter with other things such as `instanceof` thing.